product_category_code_unique/models/product_category.py

```diff
@@ -1,26 +1,73 @@
 # Copyright 2021 ACSONE SA/NV
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import api, models
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
 
 
 class ProductCategory(models.Model):
 
     _inherit = "product.category"
 
-    _sql_constraints = [
-        ("uniq_code", "unique(code)", "The category code must be unique!"),
-    ]
+    def _get_parents(self):
+        self.ensure_one()
+        return self.search([("parent_id", "parent_of", self.id)])
+
+    def _get_children(self):
+        self.ensure_one()
+        return self.search([("parent_id", "child_of", self.id)])
+
+    def _get_hierarchy_cats(self):
```

Comment:
Prefer explicit naming instead. And maybe `_get_hierarchy_categories_for_code_unique()` to avoid naming collisions.